sycl/source/detail/platform_impl.cpp

```diff
@@ -334,6 +336,17 @@ bool platform_impl::has(aspect Aspect) const {
   return true;
 }
 
+std::shared_ptr<device_impl> platform_impl::getDeviceImplHelper(
+    RT::PiDevice PiDevice, const std::shared_ptr<platform_impl> &PlatformImpl) {
+  for (const std::weak_ptr<device_impl> &DeviceWP : MDeviceCache) {
```

Comment:
A comment saying that this function is expected to be called with `MDeviceMapMutex` locked would be nice.

Reply:
Agreed, I'll add it in a separate PR.